File: Pertemuan-12/AppPerpus/Petugas.php
<?php
//Simpanlah dengan nama file : Petugas.php
require_once 'database.php';
class Petugas 
{
    private $db;
    private $table = 'petugas';
    public $nipp = "";
    public $nama = "";
    public $jabatan = "";
    public $notelp = "";
    public $alamat = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_nipp(int $nipp)
    {
        $query = "SELECT * FROM $this->table WHERE nipp = $nipp";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`nipp`,`nama`,`jabatan`,`notelp`,`alamat`) VALUES ('$this->nipp','$this->nama','$this->jabatan','$this->notelp','$this->alamat')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET nipp = '$this->nipp', nama = '$this->nama', jabatan = '$this->jabatan', notelp = '$this->notelp', alamat = '$this->alamat' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_nipp($nipp): int
    {
        $query = "UPDATE $this->table SET nipp = '$this->nipp', nama = '$this->nama', jabatan = '$this->jabatan', notelp = '$this->notelp', alamat = '$this->alamat' 
        WHERE nipp = $nipp";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_nipp($nipp): int
    {
        $query = "DELETE FROM $this->table WHERE nipp = $nipp";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Pertemuan-13/Tugas/AppPerpus/Petugas.py
import requests
import json
class Petugas:
    def __init__(self):
        self.__id=None
        self.__nipp = None
        self.__nama = None
        self.__jabatan = None
        self.__notelp = None
        self.__alamat = None
        self.__url = "http://f0832375.xsph.ru/app/petugas_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def nipp(self):
        return self.__nipp
        
    @nipp.setter
    def nipp(self, value):
        self.__nipp = value
    @property
    def nama(self):
        return self.__nama
        
    @nama.setter
    def nama(self, value):
        self.__nama = value
    @property
    def jabatan(self):
        return self.__jabatan
        
    @jabatan.setter
    def jabatan(self, value):
        self.__jabatan = value
    @property
    def notelp(self):
        return self.__notelp
        
    @notelp.setter
    def notelp(self, value):
        self.__notelp = value
    @property
    def alamat(self):
        return self.__alamat
        
    @alamat.setter
    def alamat(self, value):
        self.__alamat = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_nipp(self, nipp):
        url = self.__url+"?nipp="+nipp
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__nipp = item['nipp']
            self.__nama = item['nama']
            self.__jabatan = item['jabatan']
            self.__notelp = item['notelp']
            self.__alamat = item['alamat']
        return data
    def simpan(self):
        payload = {
            "nipp":self.__nipp,
            "nama":self.__nama,
            "jabatan":self.__jabatan,
            "notelp":self.__notelp,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_nipp(self, nipp):
        url = self.__url+"?nipp="+nipp
        payload = {
            "nipp":self.__nipp,
            "nama":self.__nama,
            "jabatan":self.__jabatan,
            "notelp":self.__notelp,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_nipp(self,nipp):
        url = self.__url+"?nipp="+nipp
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text